hack/generator/pkg/jsonast/schema_abstraction.go

```diff
@@ -66,9 +68,7 @@ type Schema interface {
 
 	// ref things
 	isRef() bool
-	refGroupName() (string, error)
-	refObjectName() (string, error)
-	refVersion() (string, error)
+	refTypeName() (astmodel.TypeName, error)
```

Comment:
Changed from separate parts to a combined `TypeName` as this makes it a lot easier to carry around `LocalPackaReference`s rather than separate `group`/`version` and combine them later. The latter was rather error-prone as well.